clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```
@@ -19,56 +19,91 @@
+     * incremented as the member reconciles and acknowledges the assignments it receives.
+     */
     private int memberEpoch;
+
+    /**
+     * Current state of this member a part of the consumer group, as defined in {@link MemberState}
+     */
     private MemberState state;
+
+    /**
+     * Assignor type selection for the member. If non-null, the member will send its selection to
+     * the server on the {@link ConsumerGroupHeartbeatRequest}. If null, the server will select a
+     * default assignor for the member, which the member does not need to track.
+     */
     private AssignorSelection assignorSelection;
 
     /**
      * Assignment that the member received from the server and successfully processed.
      */
     private ConsumerGroupHeartbeatResponseData.Assignment currentAssignment;
+
     /**
      * Assignment that the member received from the server but hasn't completely processed
      * yet.
      */
     private Optional<ConsumerGroupHeartbeatResponseData.Assignment> targetAssignment;
+
     /**
-     * Latest assignment that the member received from the server while a {@link #targetAssignment}
-     * was in process.
+     * slf4j logger.
      */
-    private Optional<ConsumerGroupHeartbeatResponseData.Assignment> nextTargetAssignment;
+    private final Logger log;
 
-    public MembershipManagerImpl(String groupId) {
-        this(groupId, null, null);
+    public MembershipManagerImpl(String groupId, LogContext logContext) {
+        this(groupId, null, null, logContext);
     }
 
-    public MembershipManagerImpl(String groupId, String groupInstanceId, AssignorSelection assignorSelection) {
+    public MembershipManagerImpl(String groupId, String groupInstanceId,
+                                 AssignorSelection assignorSelection, LogContext logContext) {
+        if (groupId == null) {
+            throw new IllegalArgumentException("Group ID cannot be null.");
```

I like this extra level of paranoia 😄  This is primarily to catch a programming issue for us though, right? A user shouldn't be able to hit this.

```suggestion
            throw new IllegalArgumentException("Consumer group membership should not be created if the group.id is null.");
```

---

It is truly to make this component solid. This is a component to manage membership to a group, it cannot do its job without a group so making sure it's explicit and fails fast. (As it is currently used, this would be a programming issue as you said, and would fail when creating this for the heartbeat manager)